code/modules/vehicles/armored/som_armored_weapons.dm

```diff
@@ -47,25 +51,39 @@
 					continue
 				if(obj_target.resistance_flags & INDESTRUCTIBLE)
 					continue
-				var/obj_damage = beam_turf ? 120 : 50
+				var/obj_damage = beam_turf ? 500 : 350
 				if(isarmoredvehicle(obj_target) || ishitbox(obj_target))
-					obj_damage *= 0.25 //you can hit them a bunch of times
-				if(ismecha(obj_target))
-					obj_damage *= 3
-				obj_target.take_damage(obj_damage, BURN, beam_turf ? ENERGY : FIRE, FALSE, attack_dir, armor_pen, current_firer)
-				return
+					obj_damage *= 0.75
+					beam_stop = TRUE
+				else if(ismecha(obj_target))
+					beam_stop = TRUE
+				obj_target.take_damage(obj_damage * 0.5, BURN, ENERGY, TRUE, attack_dir, armor_pen, current_firer)
+				if(!QDELETED(obj_target))
+					obj_target.take_damage(obj_damage * 0.5, BURN, FIRE, FALSE, attack_dir, armor_pen, current_firer)
+				continue
 			if(isliving(target))
 				var/mob/living/living_target = target
-				if(beam_turf)
-					living_target.apply_damage(50, BURN, blocked = ENERGY, penetration = armor_pen)
-				living_target.apply_damage(50, BURN, blocked = FIRE, penetration = armor_pen, updating_health = TRUE)
+				living_target.apply_damage(80, BURN, blocked = ENERGY, penetration = armor_pen)
+				if(!QDELETED(living_target))
+					living_target.apply_damage(80, BURN, blocked = FIRE, penetration = armor_pen, updating_health = TRUE)
+				living_target.flash_act(beam_turf ? 2 SECONDS : 1 SECONDS)
```

Comment:
You wanna be doing all this other stuff even when target is qdel'd?

Reply:
Yes, since we still ignite etc, although this is probably redundant since simple damage won't qdel mobs.